packages/terra-alert/src/Alert.jsx

```diff
@@ -0,0 +1,268 @@
+   */
+  title: PropTypes.string,
+  /**
+   * The type of alert to be rendered. One of Alert.Types.ALERT, Alert.Types.ERROR, Alert.Types.WARNING, Alert.Types.ADVISORY,
+   * Alert.Types.INFO, Alert.Types.SUCCESS, Alert.Types.GAP_CHECKING, Alert.Types.OUTSIDE_RECORDS, Alert.Types.CUSTOM.
+   * Use the Types attribute of the Alert component for access to these type strings.
+   */
+  type: PropTypes.oneOf([
+    AlertTypes.ALERT,
+    AlertTypes.ERROR,
+    AlertTypes.WARNING,
+    AlertTypes.ADVISORY,
+    AlertTypes.INFO,
+    AlertTypes.SUCCESS,
+    AlertTypes.GAP_CHECKING,
+    AlertTypes.OUTSIDE_RECORDS,
+    AlertTypes.CUSTOM,
+  ]),
+};
+
+const defaultProps = {
+  alertAction: null,
+  children: '',
+  customIcon: null,
+  customStatusColor: '',
+  onDismiss: null,
+  title: '',
+  type: AlertTypes.ALERT,
+};
+
+const getAlertClassName = (type) => {
+  switch (type) {
+    case AlertTypes.ALERT:
+      return 'terra-Alert--alert';
+    case AlertTypes.ERROR:
+      return 'terra-Alert--error';
+    case AlertTypes.WARNING:
+      return 'terra-Alert--warning';
+    case AlertTypes.ADVISORY:
+      return 'terra-Alert--advisory';
+    case AlertTypes.INFO:
+      return 'terra-Alert--info';
+    case AlertTypes.SUCCESS:
+      return 'terra-Alert--success';
+    case AlertTypes.GAP_CHECKING:
+      return 'terra-Alert--gap-checking';
+    case AlertTypes.OUTSIDE_RECORDS:
+      return 'terra-Alert--outside-records';
+    case AlertTypes.CUSTOM:
+      return 'terra-Alert--custom';
+    default:
+      return '';
+  }
+};
+
+const getAlertIcon = (type, customIcon) => {
+  switch (type) {
+    case AlertTypes.ALERT:
+      return (<IconAlert height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.ERROR:
+      return (<IconError height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.WARNING:
+      return (<IconWarning height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.ADVISORY:
+      return null;
+    case AlertTypes.INFO:
+      return (<IconInformation height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.SUCCESS:
+      return (<IconSuccess height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.GAP_CHECKING:
+      return (<IconRequired height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.OUTSIDE_RECORDS:
+      return (<IconDiamond height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.CUSTOM:
+      return customIcon;
+    default:
+      return null;
+  }
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+const Alert = (
+  {
+    type,
+    children,
+    title,
+    customIcon,
+    customStatusColor,
+    onDismiss,
+    alertAction,
+    ...customProps
+  }, {
+    intl,
+  },
+) => {
+  let defaultTitle = '';
+  switch (type) {
```

Comment:
It might be nice to pull this out into a function `getDefaultTitle` like you did with `getAlertIcon`